### PR TITLE
source_to_site calculation (inc. rrup) as a separate step to compute empirical

### PR DESCRIPTION
**Pull request by sungeunbae**
Previously, source_to_site calculation was done on-the-fly as a part of empirical calculation.

However, there was a need for having a step to add uncertainty to the rrup values before empirical calculation, and the current workflow was too tightly integrated. 

In this PR, the source_to_site calculation has been taken out a separate step. 

The new collect_sites_info script takes multiple input files, .ll, .vs30, .z and srf (or .nhm), and generates a single CSV file that also includes rrup, rjb, rx,ry.

This single CSV file is now a required input for the caclulate_empirical.py, making its command interface simplified.

If a researcher wishes to add perturbation to any of these values (eg. vs30, rrup, z etc), he/she can process the CSV file and feed the updated CSV file to caclulate_empirical.py

Usage:
```
(ringo310) seb56@c015kr:~/Empirical_Engine$ python ~/Empirical_Engine/empirical/scripts/collect_sites_info.py --help
                                                                                                                                      
 Usage: collect_sites_info.py [OPTIONS] SOURCE_FFP LL_FFP VS30_FFP Z_FFP                                                              
                                                                                                                                      
 Calculate source-site distances for stations in ll_ffp and put them alongside vs30 and z values into a single CSV file. Can work     
 either with SRF or NHM for a valid fault                                                                                             
                                                                                                                                      
╭─ Arguments ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ *    source_ffp      PATH  Path to the SRF or NHM file. NHM file can be obtained from                                              │
│                            https://github.com/ucgmsim/Empirical_Engine/files/15256612/NZ_FLTmodel_2010_v18p6.txt                   │
│                            [default: None]                                                                                         │
│                            [required]                                                                                              │
│ *    ll_ffp          PATH  Path to the .ll file [default: None] [required]                                                         │
│ *    vs30_ffp        PATH  Path to the .vs30 file. The file should have columns: station, vs30, (sigma) [default: None] [required] │
│ *    z_ffp           PATH  Path to the .z file that contains Z1.0 and Z2.5. The file should have columns: station, z1p0, z2p5      │
│                            [default: None]                                                                                         │
│                            [required]                                                                                              │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Options ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --out-dir                   PATH   output directory [default: /home/seb56/Empirical_Engine]                                        │
│ --nhm-fault-name            TEXT   Fault name to look up from NHM. Required if source-ffp is an NHM file                           │
│ --rjb-max                   FLOAT  Only calculate for stations whose rjb is below this value [default: 200]                        │
│ --install-completion               Install completion for the current shell.                                                       │
│ --show-completion                  Show completion for the current shell, to copy it or customize the installation.                │
│ --help                             Show this message and exit.                                                                     │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

```
Example:

Using SRF
```
python ~/Empirical_Engine/empirical/scripts/collect_sites_info.py Srf/3460014_REL01.srf geoNet_stats+2022-06-14.ll geoNet_stats+2022-06-14.vs30 geoNet_stats+2022-06-14.z 

```
Using NHM (if no SRF is available, but it is a NHM fault)

```
 python ~/Empirical_Engine/empirical/scripts/collect_sites_info.py ./NZ_FLTmodel_2010_v18p6.txt geoNet_stats+2022-06-14.ll geoNet_stats+2022-06-14.vs30 geoNet_stats+2022-06-14.z  --nhm-fault-name Ashley

```

### CODE REVIEW

File: empirical/scripts/collect_sites_info.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        float,
+        typer.Option(help="Only calculate for stations whose rjb is below this value"),
+    ] = RJB_MAX,
+):
+    """
+    Calculate source-site distances for stations in ll_ffp and put them alongside vs30 and z values into a single CSV file.
+    Can work either with SRF or NHM for a valid fault
+
+    """
+    print(source_ffp.suffix)
```
Comment (claudio525) [inline empirical/scripts/collect_sites_info.py:77]: debug print?

File: empirical/scripts/collect_sites_info.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            event_name = nhm_fault_name  # use this fault name as the event_name
+
+    out_dir.mkdir(exist_ok=True)
+
+    stations_df = formats.load_station_file(ll_ffp)
+    vs30_df = formats.load_vs30_file(vs30_ffp)
+    z_df = formats.load_z_file(z_ffp)
+    z_df = z_df.rename(columns={"z1p0": "z1pt0", "z2p5": "z2pt5"})
+
+    site_df = pd.concat([stations_df, vs30_df, z_df], axis=1)
```
Comment (claudio525) [inline empirical/scripts/collect_sites_info.py:110]: If data is missing in one of these for a site, you'll end up with NaN entries, which may cause issues down the line. Might be a good idea to add a step somewhere that checks that these agree, or if not print a message and drop those sites?

File: empirical/scripts/calculate_empirical.py [hunk 3]
```diff
@@ -25,16 +24,12 @@
...
 def run_emp(
     output_dir: Path,
-    ll_ffp: Path,
-    vs30_ffp: Path,
-    z_ffp: Path,
-    srf_ffp: Path,
-    nhm_ffp: Path = None,
+    event_name: str,
+    sites_info_ffp: Path,
     srfdata_ffp: Path = None,
```
Comment (lispandfound) [inline empirical/scripts/calculate_empirical.py:29]:
```suggestion
    srfdata_ffp: Optional[Path] = None,
```

File: empirical/scripts/collect_sites_info.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+app = typer.Typer()
+
+
+@app.command()
+def collect_sites_info_(
+    source_ffp: Annotated[
+        Path,
+        typer.Argument(
+            help="Path to the SRF or NHM file. NHM file can be obtained from https://github.com/ucgmsim/Empirical_Engine/files/15256612/NZ_FLTmodel_2010_v18p6.txt",
+            file_okay=True,
```
[the remark this answers is not in the document]
Comment (lispandfound) [reply empirical/scripts/collect_sites_info.py:27]: I'd set `dir_okay=False` instead.

**Review by lispandfound — CHANGES_REQUESTED** [review]

**Review by AndrewRidden-Harper — COMMENTED** [review]
Looks good to me, noting the others' feedback

File: empirical/scripts/calculate_empirical.py (hunk 3 above)
Comment (sungeunbae) [reply empirical/scripts/calculate_empirical.py:29]: It is slightly complicated, as srfdata_ffp is "sort of" expected, and if it doesn't, at least we need nz_gmdb_source_ffp. 
Updated the docstring for clarification.

File: empirical/scripts/collect_sites_info.py [hunk 5]
```diff
@@ -55,6 +54,7 @@ def collect_sites_info_(
         typer.Option(
             help="output directory",
             dir_okay=True,
```
Comment (joelridden) [inline empirical/scripts/collect_sites_info.py:56]: Was meaning this more of a replacement as dir_okay is default True so isnt needed

File: empirical/scripts/calculate_empirical.py (hunk 3 above)
Comment (joelridden) [reply empirical/scripts/calculate_empirical.py:29]: Even then I think it's still technically Optional because sometimes you don't need to specify it

Comment (lispandfound) [reply empirical/scripts/calculate_empirical.py]: @joelridden is correct. The `Optional` annotation just says "this value can be None sometimes". This is, rather unfortunately in my opinion, the [type-theory term](https://en.wikipedia.org/wiki/Option_type) for such values. If I were type-theory dictator for the day I'd probably change this to `Nullable` or `Maybe`.

**Review by lispandfound — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Review by joelridden — APPROVED** [review]

**Pull request merged**